### ISSUE TITLE
Creating Redis Instance Fails

### ISSUE THREAD

**Issue by zacheryph** [kubedb/project#795]
K8s: K3os / k3s `v1.19.2+k3s1`
Longhorn storage
KubeDB 0.14

I have another instance of k3s running local-path-storage and kubedb 0.13-rc1 and redis created just fine. I already have other databases on this (using the v1alpha2 namespace) so cannot downgrade to 0.13.  I cannot find any other tickets relating to this though :(

Attempting to create the quickstart redis (removed storageClass, longhorn is default)
```
$  kubectl apply -f redis-test.yaml
Error from server (InternalError): error when creating "redis-test.yaml": Internal error occurred: failed calling webhook "redis.mutators.kubedb.com": the server is currently unable to handle the request
```

Api Services are functional
```
$  kg apiservice|grep kubedb|grep operator
v1alpha1.mutators.kubedb.com              kubedb-system/operator-kubedb   True        22m
v1alpha1.validators.kubedb.com            kubedb-system/operator-kubedb   True        22m
```

Operator logs
```
E1106 04:20:31.382693       1 runtime.go:76] Observed a panic: runtime error: invalid memory address or nil pointer dereference
goroutine 12367 [running]:
k8s.io/apiserver/pkg/endpoints/handlers.finishRequest.func1.1(0xc0023e3680)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:233 +0xd5
panic(0x306d300, 0x5241310)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
kubedb.dev/apimachinery/apis/kubedb/v1alpha2.(*Redis).setDefaultAffinity(0xc000271800, 0xc000271980, 0xc001b497a0, 0x0)
	/src/vendor/kubedb.dev/apimachinery/apis/kubedb/v1alpha2/redis_helpers.go:246 +0x208
kubedb.dev/apimachinery/apis/kubedb/v1alpha2.(*Redis).SetDefaults(0xc000271800, 0x0)
	/src/vendor/kubedb.dev/apimachinery/apis/kubedb/v1alpha2/redis_helpers.go:195 +0x3f1
kubedb.dev/redis/pkg/admission.setDefaultValues(0xc000271800, 0x0, 0x380, 0x34e748d, 0xa, 0x34e3a03)
	/src/vendor/kubedb.dev/redis/pkg/admission/mutator.go:129 +0xb5
kubedb.dev/redis/pkg/admission.(*RedisMutator).Admit(0xc002c0b360, 0xc002928000, 0x0)
	/src/vendor/kubedb.dev/redis/pkg/admission/mutator.go:95 +0x1c5
kmodules.xyz/webhook-runtime/registry/admissionreview/v1beta1.(*REST).Create(0xc00051ee18, 0x3a4dd20, 0xc001b494d0, 0x39ec8a0, 0xc001b495f0, 0x3608210, 0xc0024cf6d0, 0x0, 0x125b500, 0xc000aec900, ...)
	/src/vendor/kmodules.xyz/webhook-runtime/registry/admissionreview/v1beta1/admission_review.go:61 +0x4c
k8s.io/apiserver/pkg/endpoints/handlers.(*namedCreaterAdapter).Create(0xc000bda0c0, 0x3a4dd20, 0xc001b494d0, 0x0, 0x0, 0x39ec8a0, 0xc001b495f0, 0x3608210, 0xc0024cf6d0, 0xc001b825c0, ...)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/create.go:208 +0x7f
k8s.io/apiserver/pkg/endpoints/handlers.createHandler.func1.1(0x30c0da0, 0x0, 0x0, 0x0)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/create.go:145 +0x13b
k8s.io/apiserver/pkg/endpoints/handlers.createHandler.func1.2(0xc001c0eea0, 0x0, 0x1, 0x0)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/create.go:166 +0xf5
k8s.io/apiserver/pkg/endpoints/handlers.finishRequest.func1(0xc0023e3680, 0xc0023e3560, 0xc0023e3620, 0xc0023e35c0)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:241 +0x62
created by k8s.io/apiserver/pkg/endpoints/handlers.finishRequest
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:222 +0xd4

goroutine 12365 [running]:
k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1.1(0xc002492060)
	/src/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:108 +0x113
panic(0x2f0a3c0, 0xc000bda260)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
k8s.io/apiserver/pkg/endpoints/handlers.finishRequest(0x6fc23ac00, 0xc0023e3560, 0x21, 0x0, 0x0, 0x0)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:259 +0x505
k8s.io/apiserver/pkg/endpoints/handlers.createHandler.func1(0x3a3e160, 0xc00113a038, 0xc001cb4600)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/create.go:153 +0x180d
k8s.io/apiserver/pkg/endpoints.restfulCreateResource.func1(0xc001b49350, 0xc0023b4070)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/installer.go:1104 +0xe4
k8s.io/apiserver/pkg/endpoints/metrics.InstrumentRouteFunc.func1(0xc001b49350, 0xc0023b4070)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go:323 +0x255
github.com/emicklei/go-restful.(*Container).dispatch(0xc000af2990, 0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/src/vendor/github.com/emicklei/go-restful/container.go:288 +0xa84
github.com/emicklei/go-restful.(*Container).Dispatch(...)
	/src/vendor/github.com/emicklei/go-restful/container.go:199
k8s.io/apiserver/pkg/server.director.ServeHTTP(0x34e9788, 0xb, 0xc000af2990, 0xc0006b4fc0, 0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/src/vendor/k8s.io/apiserver/pkg/server/handler.go:146 +0x4bc
k8s.io/apiserver/pkg/endpoints/filters.WithAuthorization.func1(0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/filters/authorization.go:64 +0x523
net/http.HandlerFunc.ServeHTTP(0xc00089ed80, 0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/server/filters.WithMaxInFlightLimit.func1(0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/src/vendor/k8s.io/apiserver/pkg/server/filters/maxinflight.go:160 +0x4d4
net/http.HandlerFunc.ServeHTTP(0xc000af4600, 0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/endpoints/filters.WithImpersonation.func1(0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go:50 +0x22bd
net/http.HandlerFunc.ServeHTTP(0xc00089edc0, 0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/endpoints/filters.WithAuthentication.func1(0x7f1139e80638, 0xc00113a000, 0xc001cb4500)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/filters/authentication.go:70 +0x652
net/http.HandlerFunc.ServeHTTP(0xc00009ed20, 0x7f1139e80638, 0xc00113a000, 0xc001cb4500)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1(0xc002492060, 0xc00043b400, 0x3a4e520, 0xc00113a000, 0xc001cb4500)
	/src/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:113 +0xb8
created by k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP
	/src/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:99 +0x1cc

goroutine 12408 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x2f0a3c0, 0xc000bda2a0)
	/src/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0xc001077c98, 0x1, 0x1)
	/src/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x89
panic(0x2f0a3c0, 0xc000bda2a0)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP(0xc00043b400, 0x3a3e320, 0xc0023b4000, 0xc001cb4500)
	/src/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:119 +0x448
k8s.io/apiserver/pkg/server/filters.WithWaitGroup.func1(0x3a3e320, 0xc0023b4000, 0xc001cb4400)
	/src/vendor/k8s.io/apiserver/pkg/server/filters/waitgroup.go:59 +0x137
net/http.HandlerFunc.ServeHTTP(0xc000af4630, 0x3a3e320, 0xc0023b4000, 0xc001cb4400)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/endpoints/filters.WithRequestInfo.func1(0x3a3e320, 0xc0023b4000, 0xc001cb4200)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/filters/requestinfo.go:39 +0x269
net/http.HandlerFunc.ServeHTTP(0xc000af4660, 0x3a3e320, 0xc0023b4000, 0xc001cb4200)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/endpoints/filters.WithCacheControl.func1(0x3a3e320, 0xc0023b4000, 0xc001cb4200)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/filters/cachecontrol.go:31 +0xa8
net/http.HandlerFunc.ServeHTTP(0xc00043b420, 0x3a3e320, 0xc0023b4000, 0xc001cb4200)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/server/httplog.WithLogging.func1(0x3a313e0, 0xc001922000, 0xc0023c2300)
	/src/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:89 +0x2c9
net/http.HandlerFunc.ServeHTTP(0xc00043b440, 0x3a313e0, 0xc001922000, 0xc0023c2300)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/server/filters.withPanicRecovery.func1(0x3a313e0, 0xc001922000, 0xc0023c2300)
	/src/vendor/k8s.io/apiserver/pkg/server/filters/wrap.go:51 +0xe6
net/http.HandlerFunc.ServeHTTP(0xc00043b460, 0x3a313e0, 0xc001922000, 0xc0023c2300)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/server.(*APIServerHandler).ServeHTTP(0xc000af4690, 0x3a313e0, 0xc001922000, 0xc0023c2300)
	/src/vendor/k8s.io/apiserver/pkg/server/handler.go:189 +0x51
net/http.serverHandler.ServeHTTP(0xc00002ca80, 0x3a313e0, 0xc001922000, 0xc0023c2300)
	/usr/local/go/src/net/http/server.go:2843 +0xa3
net/http.initALPNRequest.ServeHTTP(0x3a4dd20, 0xc000f2ec90, 0xc002b20000, 0xc00002ca80, 0x3a313e0, 0xc001922000, 0xc0023c2300)
	/usr/local/go/src/net/http/server.go:3415 +0x8d
golang.org/x/net/http2.(*serverConn).runHandler(0xc001254600, 0xc001922000, 0xc0023c2300, 0xc001d5ec80)
	/src/vendor/golang.org/x/net/http2/server.go:2147 +0x8b
created by golang.org/x/net/http2.(*serverConn).processHeaders
	/src/vendor/golang.org/x/net/http2/server.go:1881 +0x505
E1106 04:20:31.382711       1 wrap.go:39] apiserver panic'd on POST /apis/mutators.kubedb.com/v1alpha1/redises?timeout=30s
I1106 04:20:31.382756       1 log.go:181] http2: panic serving 10.42.1.1:10775: runtime error: invalid memory address or nil pointer dereference
goroutine 12367 [running]:
k8s.io/apiserver/pkg/endpoints/handlers.finishRequest.func1.1(0xc0023e3680)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:233 +0xd5
panic(0x306d300, 0x5241310)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
kubedb.dev/apimachinery/apis/kubedb/v1alpha2.(*Redis).setDefaultAffinity(0xc000271800, 0xc000271980, 0xc001b497a0, 0x0)
	/src/vendor/kubedb.dev/apimachinery/apis/kubedb/v1alpha2/redis_helpers.go:246 +0x208
kubedb.dev/apimachinery/apis/kubedb/v1alpha2.(*Redis).SetDefaults(0xc000271800, 0x0)
	/src/vendor/kubedb.dev/apimachinery/apis/kubedb/v1alpha2/redis_helpers.go:195 +0x3f1
kubedb.dev/redis/pkg/admission.setDefaultValues(0xc000271800, 0x0, 0x380, 0x34e748d, 0xa, 0x34e3a03)
	/src/vendor/kubedb.dev/redis/pkg/admission/mutator.go:129 +0xb5
kubedb.dev/redis/pkg/admission.(*RedisMutator).Admit(0xc002c0b360, 0xc002928000, 0x0)
	/src/vendor/kubedb.dev/redis/pkg/admission/mutator.go:95 +0x1c5
kmodules.xyz/webhook-runtime/registry/admissionreview/v1beta1.(*REST).Create(0xc00051ee18, 0x3a4dd20, 0xc001b494d0, 0x39ec8a0, 0xc001b495f0, 0x3608210, 0xc0024cf6d0, 0x0, 0x125b500, 0xc000aec900, ...)
	/src/vendor/kmodules.xyz/webhook-runtime/registry/admissionreview/v1beta1/admission_review.go:61 +0x4c
k8s.io/apiserver/pkg/endpoints/handlers.(*namedCreaterAdapter).Create(0xc000bda0c0, 0x3a4dd20, 0xc001b494d0, 0x0, 0x0, 0x39ec8a0, 0xc001b495f0, 0x3608210, 0xc0024cf6d0, 0xc001b825c0, ...)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/create.go:208 +0x7f
k8s.io/apiserver/pkg/endpoints/handlers.createHandler.func1.1(0x30c0da0, 0x0, 0x0, 0x0)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/create.go:145 +0x13b
k8s.io/apiserver/pkg/endpoints/handlers.createHandler.func1.2(0xc001c0eea0, 0x0, 0x1, 0x0)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/create.go:166 +0xf5
k8s.io/apiserver/pkg/endpoints/handlers.finishRequest.func1(0xc0023e3680, 0xc0023e3560, 0xc0023e3620, 0xc0023e35c0)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:241 +0x62
created by k8s.io/apiserver/pkg/endpoints/handlers.finishRequest
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:222 +0xd4

goroutine 12365 [running]:
k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1.1(0xc002492060)
	/src/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:108 +0x113
panic(0x2f0a3c0, 0xc000bda260)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
k8s.io/apiserver/pkg/endpoints/handlers.finishRequest(0x6fc23ac00, 0xc0023e3560, 0x21, 0x0, 0x0, 0x0)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/rest.go:259 +0x505
k8s.io/apiserver/pkg/endpoints/handlers.createHandler.func1(0x3a3e160, 0xc00113a038, 0xc001cb4600)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/handlers/create.go:153 +0x180d
k8s.io/apiserver/pkg/endpoints.restfulCreateResource.func1(0xc001b49350, 0xc0023b4070)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/installer.go:1104 +0xe4
k8s.io/apiserver/pkg/endpoints/metrics.InstrumentRouteFunc.func1(0xc001b49350, 0xc0023b4070)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go:323 +0x255
github.com/emicklei/go-restful.(*Container).dispatch(0xc000af2990, 0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/src/vendor/github.com/emicklei/go-restful/container.go:288 +0xa84
github.com/emicklei/go-restful.(*Container).Dispatch(...)
	/src/vendor/github.com/emicklei/go-restful/container.go:199
k8s.io/apiserver/pkg/server.director.ServeHTTP(0x34e9788, 0xb, 0xc000af2990, 0xc0006b4fc0, 0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/src/vendor/k8s.io/apiserver/pkg/server/handler.go:146 +0x4bc
k8s.io/apiserver/pkg/endpoints/filters.WithAuthorization.func1(0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/filters/authorization.go:64 +0x523
net/http.HandlerFunc.ServeHTTP(0xc00089ed80, 0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/server/filters.WithMaxInFlightLimit.func1(0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/src/vendor/k8s.io/apiserver/pkg/server/filters/maxinflight.go:160 +0x4d4
net/http.HandlerFunc.ServeHTTP(0xc000af4600, 0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/endpoints/filters.WithImpersonation.func1(0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go:50 +0x22bd
net/http.HandlerFunc.ServeHTTP(0xc00089edc0, 0x7f1139e80638, 0xc00113a000, 0xc001cb4600)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/endpoints/filters.WithAuthentication.func1(0x7f1139e80638, 0xc00113a000, 0xc001cb4500)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/filters/authentication.go:70 +0x652
net/http.HandlerFunc.ServeHTTP(0xc00009ed20, 0x7f1139e80638, 0xc00113a000, 0xc001cb4500)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1(0xc002492060, 0xc00043b400, 0x3a4e520, 0xc00113a000, 0xc001cb4500)
	/src/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:113 +0xb8
created by k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP
	/src/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:99 +0x1cc

goroutine 12408 [running]:
golang.org/x/net/http2.(*serverConn).runHandler.func1(0xc001922000, 0xc001077f8e, 0xc001254600)
	/src/vendor/golang.org/x/net/http2/server.go:2140 +0x16f
panic(0x2f0a3c0, 0xc000bda2a0)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0xc001077c98, 0x1, 0x1)
	/src/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x2f0a3c0, 0xc000bda2a0)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP(0xc00043b400, 0x3a3e320, 0xc0023b4000, 0xc001cb4500)
	/src/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:119 +0x448
k8s.io/apiserver/pkg/server/filters.WithWaitGroup.func1(0x3a3e320, 0xc0023b4000, 0xc001cb4400)
	/src/vendor/k8s.io/apiserver/pkg/server/filters/waitgroup.go:59 +0x137
net/http.HandlerFunc.ServeHTTP(0xc000af4630, 0x3a3e320, 0xc0023b4000, 0xc001cb4400)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/endpoints/filters.WithRequestInfo.func1(0x3a3e320, 0xc0023b4000, 0xc001cb4200)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/filters/requestinfo.go:39 +0x269
net/http.HandlerFunc.ServeHTTP(0xc000af4660, 0x3a3e320, 0xc0023b4000, 0xc001cb4200)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/endpoints/filters.WithCacheControl.func1(0x3a3e320, 0xc0023b4000, 0xc001cb4200)
	/src/vendor/k8s.io/apiserver/pkg/endpoints/filters/cachecontrol.go:31 +0xa8
net/http.HandlerFunc.ServeHTTP(0xc00043b420, 0x3a3e320, 0xc0023b4000, 0xc001cb4200)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/server/httplog.WithLogging.func1(0x3a313e0, 0xc001922000, 0xc0023c2300)
	/src/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:89 +0x2c9
net/http.HandlerFunc.ServeHTTP(0xc00043b440, 0x3a313e0, 0xc001922000, 0xc0023c2300)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/server/filters.withPanicRecovery.func1(0x3a313e0, 0xc001922000, 0xc0023c2300)
	/src/vendor/k8s.io/apiserver/pkg/server/filters/wrap.go:51 +0xe6
net/http.HandlerFunc.ServeHTTP(0xc00043b460, 0x3a313e0, 0xc001922000, 0xc0023c2300)
	/usr/local/go/src/net/http/server.go:2042 +0x44
k8s.io/apiserver/pkg/server.(*APIServerHandler).ServeHTTP(0xc000af4690, 0x3a313e0, 0xc001922000, 0xc0023c2300)
	/src/vendor/k8s.io/apiserver/pkg/server/handler.go:189 +0x51
net/http.serverHandler.ServeHTTP(0xc00002ca80, 0x3a313e0, 0xc001922000, 0xc0023c2300)
	/usr/local/go/src/net/http/server.go:2843 +0xa3
net/http.initALPNRequest.ServeHTTP(0x3a4dd20, 0xc000f2ec90, 0xc002b20000, 0xc00002ca80, 0x3a313e0, 0xc001922000, 0xc0023c2300)
	/usr/local/go/src/net/http/server.go:3415 +0x8d
golang.org/x/net/http2.(*serverConn).runHandler(0xc001254600, 0xc001922000, 0xc0023c2300, 0xc001d5ec80)
	/src/vendor/golang.org/x/net/http2/server.go:2147 +0x8b
created by golang.org/x/net/http2.(*serverConn).processHeaders
	/src/vendor/golang.org/x/net/http2/server.go:1881 +0x505
```

**Comment by zacheryph** [comment]
Disabling the mutator webhooks I ran into an error with the validator webhook also
```
I1106 05:04:10.548330       1 xray.go:250] testing ValidatingWebhook using an object with GVR = kubedb.com/v1alpha2, Resource=redises
W1106 05:04:10.552625       1 xray.go:178] Attempt 1 to detect ValidatingWebhook activation failed due to Redis.kubedb.com "test-redis-for-webhook-xray" is invalid: spec.storageType: Unsupported value: "Invalid": supported values: "Durable", "Ephemeral"
I1106 05:04:10.565245       1 recorder.go:109] Event created:  operator-kubedb.1644d2b6f951ce99
panic: Redis.kubedb.com "test-redis-for-webhook-xray" is invalid: spec.storageType: Unsupported value: "Invalid": supported values: "Durable", "Ephemeral"

goroutine 326 [running]:
kubedb.dev/operator/pkg/server.completedConfig.New.func2.1(0xc0008ed230, 0xc0001e4e00, 0xc002865260)
        /src/pkg/server/server.go:278 +0x5ac
created by kubedb.dev/operator/pkg/server.completedConfig.New.func2
```

This states the `storageType` is `Invalid` even though it is clearly stated as `Durable` within the resource yaml.

**Comment by djbobyd** [comment]
I am facing the same issue with version 0.14.0. It was working fine with 0.13.0. The issue is only related to redis. MongoDB, MySQL and PostgreSQL can be created without any problems

**Comment by zacheryph** [comment]
This appears to be resolved in`v0.15.1`. Closing ticket.

**Comment by vothanhkiet** [comment]
I'm running KubeDB version 0.15.1, and still got this issue
<img width="1488" alt="Screen Shot 2020-11-16 at 09 37 02" src="https://user-images.githubusercontent.com/4302582/99206992-662fae80-27ef-11eb-8957-89ba0970c3f7.png">
